### ISSUE TITLE
Wrong header on not allowed methods

### ISSUE THREAD

**Issue by Rayne** [bcosca/fatfree-core#105]
The framework doesn't set `X-Content-Type-Options`, `X-Frame-Options`, `X-Powered-By`, `X-XSS-Protection` and `Cache-Control` on not allowed methods. I expected that at least `X-Powered-By` gets updated to [`PACKAGE`](http://fatfreeframework.com/quick-reference#PACKAGE).

**Header** without `HEAD` route:

```
# curl -I 127.0.0.1:9000
HTTP/1.1 405 Method Not Allowed
Host: 127.0.0.1:9000
Connection: close
X-Powered-By: PHP/5.5.9-1ubuntu4.14
Allow: GET
Content-type: text/html; charset=UTF-8
```

**Header** with `HEAD` route:

```
# curl -I 127.0.0.1:9000
HTTP/1.1 200 OK
Host: 127.0.0.1:9000
Connection: close
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
X-Powered-By: Fat-Free Framework
X-XSS-Protection: 1; mode=block
Cache-Control: no-cache, no-store, must-revalidate
Content-type: text/html; charset=UTF-8
```

**Code**

``` php
<?php

require 'base.php';

$f = function(){ echo time(); };

$f3 = Base::instance();
$f3->route('GET *', $f);
# $f3->route('HEAD *', $f);
$f3->run();
```

**Comment by bcosca** [comment]
I cannot reproduce this issue.

Without HEAD route:

```
HTTP/1.1 405 Method Not Allowed
Date: Sun, 08 Jan 2017 00:23:16 GMT
Server: Apache/2.4.25 (Unix) OpenSSL/1.0.2j PHP/7.0.14
Upgrade: h2
Connection: Upgrade
X-Powered-By: Fat-Free Framework
Allow: GET
X-Frame-Options: SAMEORIGIN
X-XSS-Protection: 1; mode=block
X-Content-Type-Options: nosniff
Pragma: no-cache
Cache-Control: no-cache, no-store, must-revalidate
Expires: Thu, 01 Jan 1970 00:00:00 +0000
Content-Type: text/html; charset=UTF-8
```

With HEAD route:

```
HTTP/1.1 200 OK
Date: Sun, 08 Jan 2017 00:23:33 GMT
Server: Apache/2.4.25 (Unix) OpenSSL/1.0.2j PHP/7.0.14
Upgrade: h2
Connection: Upgrade
X-Powered-By: Fat-Free Framework
X-Frame-Options: SAMEORIGIN
X-XSS-Protection: 1; mode=block
X-Content-Type-Options: nosniff
Pragma: no-cache
Cache-Control: no-cache, no-store, must-revalidate
Expires: Thu, 01 Jan 1970 00:00:00 +0000
Content-Type: text/html; charset=UTF-8
```

**Comment by Rayne** [comment]
For the record: I am able to reproduce the issue with the outdated `3.5.1` version. The problem is fixed since `3.6.0`. :+1:
